### PR TITLE
feat: Export SignedTx

### PR DESCRIPTION
**Pull request by ookami-kb**
- Exported `SignedTx` and added `encodeBase58` method to `Signature` to be able to get the signature before actually sending the transaction.
- Turned `compile` into a method instead of a separate class, as the class does nothing useful.

### CODE REVIEW

File: packages/solana/lib/src/encoder/signature.dart [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import 'package:cryptography/cryptography.dart' as crypto;
+import 'package:solana/src/base58/encode.dart';
 import 'package:solana/src/common/byte_array.dart';
 
 class Signature extends ByteArray {
   Signature.from(crypto.Signature signature) : _data = signature.bytes;
 
   final ByteArray _data;
 
+  String encodeBase58() => base58encode(_data.toList(growable: false));
```
Comment (iharobCP) [inline packages/solana/lib/src/encoder/signature.dart:10]: Lets call this `toBase58()` as do most of the libraries.

[web3](https://solana-labs.github.io/solana-web3.js/classes/PublicKey.html#toBase58) [Go](https://github.com/portto/solana-go-sdk#new-account--get-some-airdrop) [Java](https://github.com/p2p-org/solanaj/blob/f83f95b9b484b06495456d592659a1c90eb0448a/src/main/java/org/p2p/solanaj/core/PublicKey.java#L45).

Comment (iharobCP) [reply packages/solana/lib/src/encoder/signature.dart:10]: I do know that this is not the `PublicKey` class but it is basically the same idea. We should also create a public key class eventually.

**Review by iharobCP — CHANGES_REQUESTED** [review]

File: packages/solana/lib/src/encoder/signature.dart (hunk 1 above)
Comment (ookami-kb) [reply packages/solana/lib/src/encoder/signature.dart:10]: I was thinking about using the `Signature` class from `crypto` package directly instead. But it's a theme for discussion and for a different PR.

**Review by iharobCP — APPROVED** [review]

File: packages/solana/lib/src/encoder/signature.dart (hunk 1 above)
Comment (iharobCP) [reply packages/solana/lib/src/encoder/signature.dart]: Yes, probably a good call. We could add `toBase58()` as an extension only.

**Pull request merged**